src/PublicKey.h

```diff
@@ -6,21 +6,50 @@
 
 #pragma once
 
+#include "Data.h"
+
 #include <array>
 #include <vector>
 
 namespace TW {
 
+enum class PublicKeyType {
+    secp256k1,
+    secp256k1Extended,
+    ed25519
+};
+
 class PublicKey {
 public:
-    /// The number of bytes in a compressed public key.
-    static const size_t compressedSize = 33;
+    /// The number of bytes in a secp256k1 public key.
+    static const size_t secp256k1Size = 33;
+
+    /// The number of bytes in a ed25519 public key.
+    static const size_t ed25519Size = 33;
 
-    /// The number of bytes in an uncompressed public key.
-    static const size_t uncompressedSize = 65;
+    /// The number of bytes in a secp256k1 extended public key.
+    static const size_t secp256k1ExtendedSize = 65;
 
     /// The public key bytes.
-    std::array<uint8_t, uncompressedSize> bytes;
+    Data bytes;
+
+    /// Type of public key.
+    PublicKeyType type() const {
+        switch (bytes.at(0)) {
+        case 1:
```

Comment:
each public key has first byte as curve index?

Reply:
where do I find docs on this?

Reply:
I took this from how trezor crypto does it: bip32.c line 548